### PR TITLE
Travis, 4.0 API fixes

### LINKED ISSUE #215: Updates required for SS4 compatibility
I've tweaked the builds a little to get them to run in a branch - see example failures here: https://travis-ci.org/robbieaverill/silverstripe-comments/jobs/259309989

Looks like:

* Localisations in provided permissions should not be tested (believe we had the same issue in the 3.6 builds)
* `create_tag` method needs replacement
* Request data missing from GridField tests
* Core `getTempFolder` is gone, needs replacement using a manifest probably
* Hopefully the assertions that are failing otherwise will pass once the module is updated to fix these things

### LINKED ISSUE #217: Method create_tag does not exist on FormField when trying to view comments in CMS
As per 4.0.0 changelog...
* FormField::create_tag() has been moved to `HTML` class and renamed to createTag. Invoke with
  `HTML::createTag()`.
PR fix to follow shortly.
image

### PR DESCRIPTION
**Pull request by wilr**
Fixed #215 and fixed #217

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,47 +1,34 @@
...
-script:
-  # Execute tests with no coverage.  This is the fastest option
-  - "if [ \"$COVERAGE\" != \"1\" ]; then vendor/bin/phpunit comments/tests/; fi"
+  # Bootstrap cms / mysite folder
+  - php ./cms/tests/bootstrap/mysite.php
 
-  # Execute tests with coverage.  Do this for a small
-  - "if [ \"$COVERAGE\" = \"1\" ]; then vendor/bin/phpunit --coverage-clover=coverage.clover comments/tests/; fi"
+script:
+  - if [[ $PHPUNIT_TEST ]]; then vendor/bin/phpunit; fi
```
Comment (dhensby) [inline .travis.yml]: You need to pass the directory with the tests in

Comment (wilr) [reply .travis.yml]: Ah.. I was basing it off https://github.com/silverstripe/silverstripe-multiform/blob/master/.travis.yml#L30-L35 which doesn't seem to pass anything in unless I'm missing some flag

Comment (robbieaverill) [reply .travis.yml]: It's fine to omit as long as it's defined in `phpunit.xml.dist`

File: .travis.yml [hunk 2]
```diff
@@ -1,47 +1,34 @@
...
+  - if [[ $PHPUNIT_COVERAGE_TEST ]]; then phpdbg -qrr vendor/bin/phpunit --coverage-clover=coverage.xml; fi
+  - if [[ $PHPCS_TEST ]]; then vendor/bin/phpcs --standard=framework/phpcs.xml.dist src/ tests/ ; fi
 
 after_success:
-    - "if [ \"$COVERAGE\" = \"1\" ]; then mv coverage.clover ~/build/$TRAVIS_REPO_SLUG/; fi"
-    - cd ~/build/$TRAVIS_REPO_SLUG
-    - "if [ \"$COVERAGE\" = \"1\" ]; then travis_retry codecov; fi"
+  - if [[ $PHPUNIT_COVERAGE_TEST ]]; then bash <(curl -s https://codecov.io/bash) -f coverage.xml; fi
```
Comment (robbieaverill) [inline .travis.yml]: With codecov there's a codecov.yml file that disables the PR comments which would be good to add

We'll also need to introduce the php-codesniffer composer dev requirement, and add minimum stability dev and prefer stable true to composer.json so this new style of Travis config will run.

I've started work on this here for reference: https://github.com/robbieaverill/silverstripe-comments/commits/pulls/3.0/fix-travis-config

File: webpack.config.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      bundle: Path.resolve('scss') + '/comments.scss',
+    },
+    output: {
+      path: Path.resolve('css'),
+      filename: 'styles/[name].css',
+    },
+    devtool: (ENV !== 'production') ? 'source-map' : '',
+    module: moduleCSS(ENV, PATHS),
+    plugins: pluginCSS(ENV, PATHS),
+  }
```
Comment (robbieaverill) [inline webpack.config.js]: If we're not building JS at all then we can remove most of the npm dependencies right?

File: .travis.yml [hunk 4]
```diff
@@ -1,47 +1,34 @@
...
-  - 7.0
-  - 7.1
-
-before_install:
-  - pip install --user codecov
-
-env:
-  global:
-    - DB=MYSQL CORE_RELEASE=4
+dist: precise
```
Comment (robbieaverill) [inline .travis.yml]: For SS4 we should definitely be trying to use trusty if possible since precise is being phased out

**Pull request merged**